src/mainLoop.ts

```diff
@@ -0,0 +1,51 @@
+import { Logger } from '@map-colonies/js-logger';
+import { BBox2d, Feature } from '@turf/helpers/dist/js/lib/geojson';
+import { IConfig } from 'config';
+import { container } from 'tsyringe';
+import polygonToBBox from '@turf/bbox';
+import { Services } from './common/constants';
+import { IInput } from './common/interfaces';
+import { GeoHash } from './geohash/geohash';
+import { Gpkg } from './gpkg';
+import { Worker } from './worker';
+import { intersect } from './common/utils';
+
+export class MainLoop {
+  private readonly config: IConfig;
+  private readonly logger: Logger;
+  private db?: Gpkg;
+  private worker?: Worker;
+  private readonly geohash: GeoHash;
+  private readonly input: IInput;
+
+  public constructor(input: IInput) {
+    this.config = container.resolve(Services.CONFIG);
+    this.logger = container.resolve(Services.LOGGER);
+    this.geohash = new GeoHash();
+    this.input = input;
+  }
+
+  public async run(): Promise<void> {
+    const gpkgPath = this.config.get<string>('gpkg.path');
+    const gpkgName = this.config.get<string>('gpkg.name');
+    const gpkgFullPath = `${gpkgPath}/${gpkgName}.gpkg`;
+
+    const intersection = intersect(this.input.footprint, this.input.bbox);
```

Comment:
didn't we said that the input should be geometry  or bbox?

Reply:
The input for the intersection should be footprint & bbox.
BBOX comes from user while footrpint is retrieved from the catalog.

Reply:
yes just add todo that this can be refactored to 2 geometries not just BBOX